services/frontend-service/src/ui/components/ReleaseTrainPrognosis/ReleaseTrainPrognosis.tsx

```diff
@@ -0,0 +1,211 @@
+
+    if (Object.keys(envPrognoses).length === 0) {
+        return <h1>Release train is empty</h1>;
+    }
+
+    return (
+        <div>
+            <TopAppBar showAppFilter={false} showTeamFilter={false} showWarningFilter={false} />
+            <main className="main-content commit-page">
+                {Object.entries(envPrognoses)
+                    .sort(([envName1, _1], [envName2, _2]) => envName1.localeCompare(envName2))
+                    .map(([envName, envPrognosis]) => (
+                        <EnvPrognosis key={envName} envName={envName} envPrognosis={envPrognosis} />
+                    ))}
+            </main>
+        </div>
+    );
+};
+
+const EnvPrognosis: React.FC<{ envName: string; envPrognosis: ReleaseTrainEnvPrognosis }> = ({
+    envName,
+    envPrognosis,
+}) => {
+    const outcome = envPrognosis.outcome;
+
+    let content: JSX.Element;
+
+    if (outcome === undefined) {
+        content = <p>Error retrieving the prognosis for this environment: backend returned undefined value.</p>;
+    } else {
+        if (outcome.$case === 'skipCause') {
+            content = <EnvPrognosisOutcomeSkipped skipCause={outcome.skipCause} />;
+        } else {
+            content = <EnvPrognosisOutcomeAppPrognoses appsPrognoses={outcome.appsPrognoses} />;
+        }
+    }
+
+    return (
+        <div>
+            <h1>Prognosis for release train on environment {envName}</h1>
+            {content}
+        </div>
+    );
+};
+
+const EnvPrognosisOutcomeSkipped: React.FC<{ skipCause: ReleaseTrainEnvSkipCause }> = ({ skipCause }) => {
+    switch (skipCause) {
+        case ReleaseTrainEnvSkipCause.ENV_IS_LOCKED:
+            return <p>Release train on this environment is skipped because it is locked.</p>;
+
+        case ReleaseTrainEnvSkipCause.ENV_HAS_BOTH_UPSTREAM_LATEST_AND_UPSTREAM_ENV:
+            return (
+                <p>
+                    Release train on this environment is skipped because it both has an upstream environment and is set
+                    as latest.
+                </p>
+            );
+
+        case ReleaseTrainEnvSkipCause.ENV_HAS_NO_UPSTREAM:
+            return <p>Release train on this environment is skipped because it has no upstream configured.</p>;
+
+        case ReleaseTrainEnvSkipCause.ENV_HAS_NO_UPSTREAM_LATEST_OR_UPSTREAM_ENV:
+            return (
+                <p>
+                    Release train on this environment is skipped because it neither has an upstream environment
+                    configured nor is marked as latest.
+                </p>
+            );
+
+        case ReleaseTrainEnvSkipCause.UPSTREAM_ENV_CONFIG_NOT_FOUND:
+            return (
+                <p>
+                    Release train on this environment is skipped because no configuration was found for it in the
+                    manifest repository.
+                </p>
+            );
+
+        case ReleaseTrainEnvSkipCause.UNRECOGNIZED:
```

Comment:
Redundant case

Reply:
Indeed, but still I'd rather be explicit than implicit about it.